src/librustdoc/html/render/mod.rs

```diff
@@ -1039,22 +1039,29 @@ themePicker.onblur = handleThemeButtonsBlur;
         cx.shared.fs.write(&dst, v.as_bytes())?;
     }
 
-    // Update the search index
+    // Update the search index and crate list.
     let dst = cx.dst.join(&format!("search-index{}.js", cx.shared.resource_suffix));
     let (mut all_indexes, mut krates) = try_err!(collect_json(&dst, &krate.name.as_str()), &dst);
     all_indexes.push(search_index);
+    krates.push(krate.name.to_string());
+    krates.sort();
+    krates.dedup();
```

Comment:
Just wondering: is that case even possible? Normally you can only have each crate once. Did you found a case where there was a duplicate?

Reply:
I was moving this up from here: https://github.com/rust-lang/rust/pull/82310/files#diff-40a0eb025da61717b3b765ceb7fab21d91af3012360e90b9f46e15a4047946faL1081. I do think it's possible that `krates.push(krate.name.to_string())` adds a crate name that is a duplicate of a crate already in the list.

Reply:
Look at the code of `collect_json`, this more precisely:

```rust
// ...
let prefix = format!("\"{}\"", krate);
// ...
    if line.starts_with(&prefix) {
        continue;
    }
// ...
```

If the crate is already in the search index, we don't add it to the crates list returned. So the `dedup` is unnecessary here.